app/models/RideManager.py

```diff
@@ -0,0 +1,51 @@
+import random
+from functools import lru_cache
+from typing import Tuple
+
+from geopy.distance import geodesic
+
+from app.models import Customer
+from app.models.DriverManager import DriverManager
+from app.models.Ride import Ride
+
+
+class RideManager:
+    def __init__(self):
+        self.book_Details = {}
+        self.driverManager = DriverManager()
+
+        self.basic_fare = 8.0
+        self.basic_distance = 10.0
+        self.per_mile_rate = 2.5
+        self.min_fare = 13.0
+        self.long_trip_baseline = 50
+        self.long_trip_adder = 2.0
+
+    def bookRide(self, customer: Customer, source: Tuple[float, float], destination: Tuple[float, float]) -> Ride:
+        available_near_driver = self.driverManager.getClosestDriver(source)
+        tariff_for_the_ride = self.calculateTariff(source, destination)
+        new_booking = Ride(customer, available_near_driver, tariff_for_the_ride, source, destination)
+        available_near_driver.isAvailable = False
+        self.driverManager.updateDriverAvailability(available_near_driver.driverId,available_near_driver.isAvailable )
```

Comment:
Why are we updating driver availability twice ? I think this can lead to bugs in future. Try to keep this implementation cleaner. 

When picking drivers from available pool of drivers, only get the reference to the driver. Handle any modifications to a driver object only through the driver manager. 

Later when we bring in concurrency and multiple people trying to book rides, it will be an issue if individual driver objects are being modified separately from multiple points in the app.